includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1482,54 +1421,72 @@ protected function build_display_items( $itemized_display_items = false ) {
 				$subtotal      += $cart_item['line_subtotal'];
 				$quantity_label = 1 < $cart_item['quantity'] ? ' (x' . $cart_item['quantity'] . ')' : '';
 
-				$product_name = $cart_item['data']->get_name();
-
 				$item = [
-					'label'  => $product_name . $quantity_label,
+					'label'  => $cart_item['data']->get_name() . $quantity_label,
 					'amount' => WC_Stripe_Helper::get_stripe_amount( $amount ),
 				];
 
 				$items[] = $item;
 			}
 		}
 
-		if ( version_compare( WC_VERSION, '3.2', '<' ) ) {
-			$discounts = wc_format_decimal( WC()->cart->get_cart_discount_total(), WC()->cart->dp );
-		} else {
-			$applied_coupons = array_values( WC()->cart->get_coupon_discount_totals() );
+		// Tax
+		$tax              = WC()->cart->tax_total + WC()->cart->shipping_tax_total;
+		$tax_to_substract = 0;
 
-			foreach ( $applied_coupons as $amount ) {
-				$discounts += (float) $amount;
+		if ( wc_tax_enabled() ) {
+			if ( ! $has_shipping_address ) {
+				$tax_to_substract = $tax;
+				$tax              = 0;
 			}
-		}
 
-		$discounts   = wc_format_decimal( $discounts, WC()->cart->dp );
```

Comment:
I noticed you removed `wc_format_decimal` from a couple places. Is that fine?

I just wanted to ask because it didn't break for me, but I didn't test this part thoroughly.

Reply:
Thanks for asking. I couldn't find a justification to use it. The value is not stored in the DB neither is sent to the user so I went ahead and removed it.

Reply:
I've been thinking about this and decided to restore the previous usage of `wc_format_decimal` for taxes, shipping and discounts. I suspect it has to do with making a better rounding on monetary values. So, in order to not break something unintentionally I'm restoring the usage of `wc_format_decimal` as it was before. We can always remove it later if we find it's not necessary. c0769e6